appium/options/android/common/adb/adb_options.py

```diff
@@ -0,0 +1,227 @@
+# Licensed to the Software Freedom Conservancy (SFC) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The SFC licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from datetime import timedelta
+from typing import Any, TypeVar
+
+from appium.options.common.supports_capabilities import SupportsCapabilities
+
+C = TypeVar('C', bound='SupportsCapabilities')
```

Comment:
I assume this class needs must be a [generic](https://mypy.readthedocs.io/en/stable/generics.html) one

Reply:
> I assume this class needs must be a [generic](https://mypy.readthedocs.io/en/stable/generics.html) one

But then what should be the type of `obj` and `cls` in `__get__` and `__set__` method. I think it has to be of type `SupportCapabilities`.

Reply:
I actually meant the descriptor class this is needed in order to properly describe the type returned by the getter and set in the setter

Reply:
> I actually meant the descriptor class. This is needed in order to properly describe the type returned by the getter and set in the setter

Got it. Thanks! I will do this change.

Reply:
@mykola-mokhnach I have incorporated all the review comments that were given. Request you the review and let me know in case if I need to change anything. Thank You.